File: Easy/Java/Maximum_Subarray(53).java
class Solution { 
    public int maxSubArray(int[] nums) {
        int curr_sum = nums[0];
        int max_sum = curr_sum;
        
        if(nums.length == 1) return nums[0];
        
        for(int i = 1; i < nums.length; i++)
        {
            curr_sum = Math.max(curr_sum + nums[i], nums[i]);
            
            max_sum = Math.max(curr_sum, max_sum);
        }
        return max_sum;
    }
}

File: Easy/C++/Maximum_Subarray(53).cpp
class Solution {
public:
int maxSubArray(vector<int> &nums) {
// write your code here
    int cur = nums[0], sum = nums[0];
    for(int i = 1; i < nums.size(); i++){
        cur = max(nums[i], cur + nums[i]);
        sum = max(cur, sum);
    }
    return sum;
}
};